pkg/cmd/chat/file/file.go

```diff
@@ -0,0 +1,193 @@
+			$ stream-cli chat upload-image --channel-type messaging --channel-id redteam --user-id "user-1" --file "./picture.png" --content-type "image/png"
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c, err := config.GetConfig(cmd).GetClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			chType, _ := cmd.Flags().GetString("channel-type")
+			chId, _ := cmd.Flags().GetString("channel-id")
+			user, _ := cmd.Flags().GetString("user-id")
+			file, _ := cmd.Flags().GetString("file")
+			contentType, _ := cmd.Flags().GetString("content-type")
+
+			path, err := utils.UploadImage(c, cmd, chType, chId, user, file, contentType)
+			if err != nil {
+				return err
+			}
+
+			cmd.Printf("Successfully uploaded image: %s\n", path)
+			return nil
+		},
+	}
+
+	fl := cmd.Flags()
+	fl.StringP("channel-type", "t", "", "[required] Channel type to interact with")
+	fl.StringP("channel-id", "i", "", "[required] Channel id to interact with")
+	fl.StringP("user-id", "u", "", "[required] User id")
+	fl.StringP("file", "f", "", "[required] Image file path")
+	fl.StringP("content-type", "c", "", "[required] Content type of the image")
+	cmd.MarkFlagRequired("channel-type")
+	cmd.MarkFlagRequired("channel-id")
+	cmd.MarkFlagRequired("user-id")
+	cmd.MarkFlagRequired("file")
+	cmd.MarkFlagRequired("content-type")
+
+	return cmd
+}
+
+func deleteFileCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "delete-file --channel-type [channel-type] --channel-id [channel-id] --file-url [file-url]",
+		Short: "Delete a file",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c, err := config.GetConfig(cmd).GetClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			chType, _ := cmd.Flags().GetString("channel-type")
+			chId, _ := cmd.Flags().GetString("channel-id")
+			fileUrl, _ := cmd.Flags().GetString("file-url")
+
+			err = utils.DeleteFile(c, cmd, chType, chId, fileUrl)
+			if err != nil {
+				return err
+			}
+
+			cmd.Println("Successfully deleted file")
+			return nil
+		},
+	}
+
+	fl := cmd.Flags()
+	fl.StringP("channel-type", "t", "", "[required] Channel type to interact with")
+	fl.StringP("channel-id", "i", "", "[required] Channel id to interact with")
+	fl.StringP("file-url", "u", "", "[required] Url of the file to delete")
+	cmd.MarkFlagRequired("channel-type")
+	cmd.MarkFlagRequired("channel-id")
+	cmd.MarkFlagRequired("file-url")
+
+	return cmd
+}
+
+func deleteImageCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "delete-image --channel-type [channel-type] --channel-id [channel-id] --image-url [image-url]",
+		Short: "Delete an image",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c, err := config.GetConfig(cmd).GetClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			chType, _ := cmd.Flags().GetString("channel-type")
+			chId, _ := cmd.Flags().GetString("channel-id")
+			imageUrl, _ := cmd.Flags().GetString("image-url")
+
+			err = utils.DeleteImage(c, cmd, chType, chId, imageUrl)
```

Comment:
Why don't we call `c.Channel()...` directly here?

Reply:
same concerns for delete file

Reply:
true, does not make much sense.